go-ussd_registration.js

```diff
@@ -1621,7 +1635,15 @@ go.app = function() {
                             if (    contact && contact.details && contact.details.addresses &&
                                     contact.details.addresses.msisdn && contact.details.addresses.msisdn[msisdn] &&
                                     contact.details.addresses.msisdn[msisdn].optedout) {
-                                return 'state_save_identities';
+                                //return 'state_save_identities';
+                                self.im.user.set_answer('mother_id', contact.id);
+                                self.im.user.set_answer('receiver_id', contact.id);
+                                if (bypassPostbirth) {
+                                    self.im.user.set_answer('state_pregnancy_status', 'prebirth');
+                                    return self.states.create('state_last_period_month');
+                                } else {
+                                    return self.states.create('state_pregnancy_status');
+                                }
```

Comment:
We don't want to go save the identity because we found it in `get_identity_by_address` above, so we skip `state_save_identities` and carry on. It was already like this in the voice registration file and it eliminates a duplicate call to the identity store. I'll clean up the comments.